### PR TITLE
Move date_spec evaluation to libcrmcommon

### PR DESCRIPTION
**Pull request by kgaillot**

### CODE REVIEW

**Comment by kgaillot** [comment]
@nrwahl2 , here's another low-priority review. Thanks

**Comment by kgaillot** [comment]
Added one commit on top

File: lib/common/rules.c [hunk 1]
```diff
@@ -147,50 +147,72 @@ check_range(const xmlNode *date_spec, const char *attr, uint32_t value)
...
-    if (subpart_rc != pcmk_rc_ok) { \
-        return subpart_rc; \
-    } \
-} while (0)
-
+/*!
+ * \internal
+ * \brief Evaluate a date_spec for a given date/time
+ *
+ * \param[in] date_spec  XML of date_spec element to check against
```
Comment (nrwahl2) [inline lib/common/rules.c:155]: `PCMK_XE_DATE_SPEC`, ideally

**Review by nrwahl2 — COMMENTED** [review]
Nice

File: .gitignore [hunk 2]
```diff
@@ -339,7 +339,7 @@ pacemaker_*.info
 /lib/common/tests/utils/pcmk_str_is_infinity
 /lib/common/tests/utils/pcmk_str_is_minus_infinity
 /lib/gnu/libgnu.a
-/lib/pengine/tests/rules/pe_cron_range_satisfied
+/lib/pengine/tests/rules/
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply .gitignore:342]: nope, that wouldn't match `pe_cron_range_satisfied` (no `_test` -- it was one of the original unit tests before we adopted that convention)

File: lib/common/tests/rules/Makefile.am [hunk 3]
```diff
@@ -10,8 +10,6 @@
 include $(top_srcdir)/mk/tap.mk
 include $(top_srcdir)/mk/unittest.mk
 
-LDADD += $(top_builddir)/lib/pengine/libpe_rules_test.la
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/common/tests/rules/Makefile.am:13]: The relevant function was moved to libcrmcommon

File: lib/common/rules.c [hunk 4]
```diff
@@ -97,69 +98,58 @@ phase_of_the_moon(const crm_time_t *now)
...
+ * \internal
+ * \brief Check an integer value against a range from PCMK_XE_DATE_SPEC XML
+ *
+ * \param[in] date_spec  XML of PCMK_XE_DATE_SPEC element to check
+ * \param[in] attr       Name of XML attribute with range to check against
+ * \param[in] value      Value to compare against range
+ *
+ * \return Standard Pacemaker return code (specifically, pcmk_rc_before_range,
+ *         pcmk_rc_after_range, or pcmk_rc_ok to indicate that result is either
+ *         withing range or undetermined)
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/common/rules.c:112]: thanks

File: lib/common/rules.c [hunk 5]
```diff
@@ -97,69 +98,58 @@ phase_of_the_moon(const crm_time_t *now)
...
-    const char *value = crm_element_value(cron_spec, xml_field);
+    int rc = pcmk_rc_ok;
+    const char *range = crm_element_value(date_spec, attr);
     long long low, high;
 
-    if (value == NULL) {
-        /* Return pe_date_result_undetermined if the field is missing. */
-        goto bail;
+    if (range == NULL) {
+        goto bail; // Attribute not present, return pcmk_rc_undetermined
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/common/rules.c:125]: hmm, not sure how i messed that up

**Comment by kgaillot** [comment]
Updated per review

File: lib/common/rules.c (hunk 5 above)
[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/common/rules.c]: ahhh, now i remember -- i changed it so pcmk_rc_ok is also used for undetermined, to simplify the callers

**Comment by kgaillot** [comment]
reverted rc change

File: .gitignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (nrwahl2) [reply .gitignore:342]: I mean we don't include that anymore, so we shouldn't have artifacts from it... I guess if anyone has a REALLY old source tree that they haven't clenaed out...

File: lib/common/tests/rules/Makefile.am (hunk 3 above)
[the remark this answers is not in the document]
Comment (nrwahl2) [reply lib/common/tests/rules/Makefile.am:13]: Right, I figured there'd be an equivalent `LDADD` directive for that

**Review by nrwahl2 — APPROVED** [review]

File: .gitignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (kgaillot) [reply .gitignore:342]: Yeah, that whole "Formerly built" section is stuff that for leftovers from previous code. Besides simply not cleaning out old files, it helps when you checkout an old commit and rebuild it for whatever reason, then jump back to where you were.

File: lib/common/tests/rules/Makefile.am (hunk 3 above)
[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/common/tests/rules/Makefile.am:13]: Ah right, that's in unittest.mk

File: .gitignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (nrwahl2) [reply .gitignore:342]: Y'know, if I'd seen the header ("Formerly built files (helps when jumping back and forth in checkout") I wouldn't have had this question ;)

**Pull request merged**